[material-ui][PaginationItem] Add slot descriptions and default components in API documentation

sai6855 opened the pull request:
While working on https://github.com/mui/material-ui/pull/46666 , i noticed slots table is empty. Created PR to fill all missing details

<img width="483" height="254" alt="image" src="https://github.com/user-attachments/assets/ef20582d-b917-4c7a-bb31-71b0c3bb1395" />


Before : https://mui.com/material-ui/api/pagination-item/#slots

After: https://deploy-preview-46677--material-ui.netlify.app/material-ui/api/pagination-item/#slots

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
 
 export interface PaginationItemSlots {
+  /**
+   * The component that renders the root slot.
```
"root" slot is not there.
```suggestion
   * Component rendered for the "first page" button.
```

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
 export interface PaginationItemSlots {
+  /**
+   * The component that renders the root slot.
+   * @default FirstPage
```
```suggestion
   * @default FirstPageIcon
```

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
+   * @default FirstPage
+   */
   first: React.ElementType;
+  /**
+   * The component that renders the last slot.
+   * @default LastPage
+   */
```
```suggestion
  /**
   * Component rendered for the "last page" button.
   * @default LastPageIcon
   */
```

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
+   * @default LastPage
+   */
   last: React.ElementType;
+  /**
+   * The component that renders the next slot.
+   * @default NavigateNext
+   */
```
Same as above.

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
-      PaginationItemOwnerState
-    >;
+    /**
+     * Props forwarded to the first slot.
+     * By default, the available props are based on svg element.
```
```suggestion
     * Props forwarded to the "first page" slot component.
```

sai6855 replied:
Most of components are using `Props forwarded to `slotName` slot`, `component` is not present in description. check https://github.com/mui/material-ui/blob/d92ea44ed02d7770182e3d14fb2507f61ee770df/packages/mui-material/src/Avatar/Avatar.d.ts#L38

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
+     * Props forwarded to the first slot.
+     * By default, the available props are based on svg element.
+     */
+    first: SlotProps<'svg', {}, PaginationItemOwnerState>;
```
I don't think it should be just `svg`. It can render any React element, button, or custom icon. The earlier version was correct.

ZeeshanTamboli commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -14,23 +14,51 @@ export interface PaginationItemPropsSizeOverrides {}
+     * By default, the available props are based on svg element.
+     */
+    first: SlotProps<'svg', {}, PaginationItemOwnerState>;
+    /**
+     * Props forwarded to the last slot.
+     * By default, the available props are based on svg element.
+     */
```
Same as above.

sai6855 commented:
@ZeeshanTamboli 

here are summary of changes done in last 4 commits

1. Most of the slots description follows `The component that renders "slot-name" slot`, I followed same in this PR as well. check https://mui.com/material-ui/api/table-sort-label/#slots, https://mui.com/material-ui/api/speed-dial/#slots, https://mui.com/material-ui/api/avatar/#slots
2. Added `Icon` to default component name https://github.com/mui/material-ui/pull/46677/commits/c49533b598f375c69532a5172116b2fa06154dab
3. Added overrides to override default slotProps when users want to override default component

sai6855 commented on packages/mui-material/src/PaginationItem/PaginationItem.d.ts:
```diff
@@ -13,22 +13,71 @@ export interface PaginationItemPropsSizeOverrides {}
-    next: SlotProps<React.ElementType<React.HTMLProps<HTMLElement>>, {}, PaginationItemOwnerState>;
+    /**
+     * Props forwarded to the first page slot.
+     * By default, the available props are based on svg element.
```
Why? Default element is `svg` know 🤔 ?

sai6855 replied:
Why? default element is `svg` know 🤔 ?

ZeeshanTamboli replied:
It's not about the default element being an SVG — it's implying that `slotProps[slot]` only has SVG props by default. But since we reverted the change, it can now accept any element props.
